CZBANK-23 - adding two new features

vojtech-cerveny opened the pull request:
feature: adding delay 5 second in creating new transaction
feature: bug in security - authorized user can check others bank account details
fix: fix for better-auth - if API key was invalid, server returned 500

<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Adds feature flags to delay transactions and control bank-account detail visibility, refactors account lookups to enforce ownership, improves auth error handling, and adds loading state to transfer UI.
> 
> - **Backend/API**:
>   - **Bank Account routes**: `GET /api/v1/bank-account/{id}` now conditionally bypasses ownership via `CAN_SEE_YOUR_BANK_ACCOUNT_DETAIL`; otherwise uses `getBankAccountByIdAndUserId`. `DELETE`/`PATCH` enforce ownership via `getBankAccountByIdAndUserId`.
>   - **Services/Repo**: Add `repository.getBankAccountById(id)` and `service.getBankAccountById(id)`; rename `service.getBankAccountByIdAndUserId` for ownership checks.
>   - **Auth**: `checkUserAuthOrThrowError` wrapped in try/catch to return `UNAUTHORIZED` on errors instead of 500.
>   - **Transactions**: When `INCREASE_TIME_IN_SENDING_TRANSACTIONS` is enabled, add a 5s delay before processing.
> - **Features**:
>   - Add `INCREASE_TIME_IN_SENDING_TRANSACTIONS` and `CAN_SEE_YOUR_BANK_ACCOUNT_DETAIL`; expand categories for `BUG_INCORRECT_BALANCE_DISPLAY`.
> - **Frontend**:
>   - Bank account page uses `getBankAccountByIdAndUserId` for ownership enforcement.
>   - Transfer form: add loading state with spinner, disable submit during processing; improved success/failure toasts and field reset.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 498e3b42eb1672982162e4f4f428551d43468fa4. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

vojtech-cerveny committed:
feature: adding feature for "time"

- increase time for sending money.
- improve UI for button during sending money

vojtech-cerveny committed:
feat: implement feature toggle for bank account visibility

- Added a feature to allow visibility of bank account details based on user permissions.
- Updated API routes to conditionally retrieve bank account information based on the new feature.
- Introduced new service methods and repository functions to support the feature.
- Enhanced error handling and response structure for bank account retrieval.

vojtech-cerveny committed:
fix: enhance error handling in user authentication

- Fix problem with wrong API key - it returned 500

postulkam commented on src/components/transactions/transfer.tsx:
```diff
@@ -69,37 +75,47 @@ export function TransactionTransfer({
-    } else {
+
+      form.resetField("amount");
+      form.resetField("toBankNumber");
```
I think, this will be better to move to `finally` or add it to the `catch` too. In case of error, it will not reset the fields.

```
form.resetField("amount");
form.resetField("toBankNumber");
```

postulkam commented on src/domain/bankAccount-domain/ba-repository.ts:
```diff
@@ -112,6 +112,18 @@ export async function getBankAccountByIdAndUserId(bankAccountId: string, userId:
 }
 
+export async function getBankAccountById(bankAccountId: string) {
+  console.log(bankAccountId);
```
Can we remove the console log? Normally, we don't add these logs to production version. :)

postulkam commented on src/domain/transaction-domain/transaction-service.ts:
```diff
@@ -2,10 +2,14 @@ import env from "@/lib/env";
 import { CreateTransactionNumberToNumberSchema } from "./transation-schema";
 
+const allFeatures = await featuresService.server.getAllFeatures();
+
```
I agree with cursor. :D 

This should be part of the `parseTransactionData` and we should get the actual state of features every time we call the function.

postulkam reviewed (approved):
Three little comments/suggestions.

They are minor ones, it is up to you if you want to resolve them or not or just add them to some of your future PR. ;)

vojtech-cerveny committed:
fix: improve error handling and reset form fields in transaction transfer

- Enhanced error handling in the transaction transfer process to provide clearer feedback on failures.
- Moved form field reset logic to the finally block to ensure fields are reset regardless of success or failure.
